kie-soup-dataset/kie-soup-dataset-api/src/main/java/org/dashbuilder/dataset/DataSetLookup.java

```diff
@@ -54,6 +56,8 @@
      * The list of operations to apply on the target data set as part of the lookup operation.
      */
     protected List<DataSetOp> operationList = new ArrayList<DataSetOp>();
+    
+    private final Map<String, Object> metadata = new HashMap<>();
```

Comment:
@dgutierr, added a change to this PR. End up having to extend the datasetlookup to avoid using threadlocal to store start time, hopefully you're ok with it.

Reply:
@cristianonicolai This is a very generic mechanism to pass data along the lookup instance. Not sure if that's too much tbh. I guess you are using so to store the start time, so may be you can just add a createdDateTime field? Either way I'm ok with this change..